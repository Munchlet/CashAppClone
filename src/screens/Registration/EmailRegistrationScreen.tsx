import React, { useEffect, useRef, useState } from "react";
import { Keyboard, StyleSheet, Text, TextInput, View, ViewStyle } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { VirtualKeyboard } from "react-native-screen-keyboard";
import { s } from "react-native-size-matters";
import Button from "../../shared/components/Button";
import { ThemeType } from "../../shared/ThemeManager";
import SafeScreen from "./SafeScreen";
import { validateEmail, validatePhone } from "../../utils/Helper";
import { ValidateRegistrationResult } from "../../typings";

// interface Props extends StackNavigationProp {}

type SelectedMethod = "phone" | "email";

export default function EmailRegistrationScreen() {
	const keyboardRef = useRef<VirtualKeyboard>(null);
	const inputRef = useRef<TextInput>(null);
	const [method, setMethod] = useState<SelectedMethod>("phone");
	const [value, setValue] = useState<string>("");
	const [placeholder, setPlaceholder] = useState<string>("Phone or Email");
	const [disableButton, setDisableButton] = useState<boolean>(true);

	useEffect(() => {
		Keyboard.dismiss();
	}, []);

	const onKeyPressed = (key: string) => {
		switch (key) {
			case "custom":
				setMethod("email");
				setPlaceholder("Email");
				setValue("");
				if (inputRef.current) inputRef.current.focus();
				break;
			case "back":
				setValue(prev => prev.slice(0, -1));
				break;
			default:
				setValue(prev => `${prev}${key}`);
		}
	};

	const onInputChange = (text: string): void => {
		let result = validateInput(text);
		switch (method) {
			case "email":
				result = validateInput(text);
				setValue(text);
				break;
			case "phone":
				// won't happen, only used for testing
				setValue(result.parsedText ?? text);
				break;
		}

		setDisableButton(!result.success);
	};

	const onNextClick = (): void => {
		const { success } = validateInput(value);
		if (!success) return console.log("Failed");
	};

	const onUsePhoneClick = (): void => {
		setMethod("phone");
		setPlaceholder("Phone or Email");
		setValue("");
		if (inputRef.current) inputRef.current.blur();
		Keyboard.dismiss();
	};

	const validateInput = (text: string): ValidateRegistrationResult => {
		if (text.length < 1) return { success: false };
		switch (method) {
			case "phone":
				const parsedText = text.replace(/\D/g, "");
				if (validatePhone(text)) return { success: true, parsedText };
				break;
			case "email":
				if (validateEmail(text)) return { success: true };
				break;
		}

		return { success: false };
	};

	return (
		<SafeAreaView style={{ flex: 1, backgroundColor: "white", justifyContent: "space-between" }}>
			<SafeScreen style={styles.topContainer}>
				<Text style={styles.header}>Enter your phone or email</Text>
				<TextInput
					testID="test-input-email-phone"
					ref={inputRef}
					style={styles.textInput}
					onChangeText={onInputChange}
					value={value}
					placeholder={placeholder}
					// showSoftInputOnFocus={false}
					onFocus={() => {
						// for iOS?
						if (method === "phone") {
							// Keyboard.dismiss();
							return false;
						}
					}}
				/>
			</SafeScreen>

			{method === "phone" ? (
				<View style={styles.bottomContainer}>
					<VirtualKeyboard
						testID="test-virtual-keyboard"
						keyStyle={{ backgroundColor: "white", borderRightWidth: 0, borderBottomWidth: 0 }}
						style={styles.keyboard}
						onRef={(ref: any) => (keyboardRef.current = ref)}
						onKeyDown={onKeyPressed}
						// keyboard={[
						// 	[1, 2, 3],
						// 	[4, 5, 6],
						// 	[7, 8, 9],
						// 	["abc", 0, 1],
						// ]}
					/>
					<View style={{ alignItems: "center", marginTop: 20 }}>
						<Button
							testKey="phone-next"
							width={90}
							text="Next"
							onPress={onNextClick}
							disabled={disableButton}
						></Button>
					</View>
				</View>
			) : (
				<View
					style={[
						styles.bottomContainer,
						{
							flexDirection: "row",
							height: s(50),
							alignItems: "flex-end",
							paddingHorizontal: 20,
							justifyContent: "space-evenly",
						},
					]}
				>
					<View style={{ flex: 1, alignItems: "center" }}>
						<Button
							testKey="use-phone"
							width={90}
							theme={ThemeType.SECONDARY}
							text="Use Phone"
							onPress={onUsePhoneClick}
						></Button>
					</View>
					<View style={{ flex: 1, alignItems: "center" }}>
						<Button
							testKey="email-next"
							width={90}
							text="Next"
							onPress={onNextClick}
							disabled={disableButton}
						></Button>
					</View>
				</View>
			)}
		</SafeAreaView>
	);
}

interface Styles {
	header: ViewStyle;
	keyboard: ViewStyle;
	textInput: ViewStyle;
	topContainer: ViewStyle;
	bottomContainer: ViewStyle;
	button: ViewStyle;
	buttonText: ViewStyle;
	subtitle: ViewStyle;
}

const styles = StyleSheet.create<Styles>({
	header: {
		fontFamily: "GothamMed",
		fontSize: 26,
	},
	keyboard: {},
	textInput: {
		width: "100%",
		height: 30,
		fontSize: 24,
		marginTop: 20,
	},
	topContainer: {
		paddingTop: 66,
		paddingLeft: 36,
		flexBasis: 140,
		flex: 1,
	},
	bottomContainer: {
		marginBottom: 26,
	},
	button: {
		marginTop: 20,
		minWidth: "86%",
		backgroundColor: "limegreen",
		justifyContent: "center",
		alignItems: "center",
		height: s(36),
		borderRadius: 30,
	},
	buttonText: {
		fontFamily: "GothamMed",
		color: "white",
		fontSize: 18,
	},
	subtitle: {
		marginVertical: 16,
		fontSize: 18,
		color: "grey",
	},
});
