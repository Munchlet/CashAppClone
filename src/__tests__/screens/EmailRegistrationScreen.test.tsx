import React from "react";
import { fireEvent, render, cleanup } from "@testing-library/react-native";
import EmailRegistrationScreen from "../../screens/Registration/EmailRegistrationScreen";
import "@testing-library/jest-native/extend-expect";

afterEach(cleanup);

describe("<EmailRegistrationScreen />", () => {
	it("should mount component", () => {
		const { toJSON } = render(<EmailRegistrationScreen />);
		expect(toJSON()).toMatchSnapshot();
	});

	it("should properly render input and keyboard", () => {
		const { debug, getByTestId, getByText, queryByTestId, toJSON } = render(<EmailRegistrationScreen />);
		const input = getByTestId("test-input-email-phone");
		expect(input).toBeDefined();
		// const keyboard = getByTestId("test-virtual-keyboard");
		// expect(keyboard).toBeDefined();
	});

	it("should validate input", () => {
		const { debug, getByTestId, getByText, queryByTestId, toJSON } = render(<EmailRegistrationScreen />);
		const input = getByTestId("test-input-email-phone");
		fireEvent.changeText(input, "1234");
		expect(input.props.value).toBe("1234");
		fireEvent.changeText(input, "abcd");
		expect(input.props.value).toBe("");
	});

	it("should validate input", () => {
		const { debug, getByTestId, getByText, queryByTestId, toJSON } = render(<EmailRegistrationScreen />);
		const input = getByTestId("test-input-email-phone");

		const b1 = getByTestId("VirtualKeyboard-1");
		const b2 = getByTestId("VirtualKeyboard-2");
		const b3 = getByTestId("VirtualKeyboard-3");
		fireEvent.press(b1);
		fireEvent.press(b2);
		fireEvent.press(b3);
		expect(input.props.value).toBe("123");
	});
});