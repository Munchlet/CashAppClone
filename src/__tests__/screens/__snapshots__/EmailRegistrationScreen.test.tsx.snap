// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmailRegistrationScreen /> should mount component 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "paddingLeft": 77.14285714285714,
          "paddingTop": 129.4764705882353,
        },
        Object {
          "flex": 1,
          "flexBasis": 140,
          "paddingLeft": 36,
          "paddingTop": 66,
        },
      ]
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "GothamMed",
          "fontSize": 26,
        }
      }
    >
      Enter your phone or email
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder="Phone or Email"
      rejectResponderTermination={true}
      style={
        Object {
          "fontSize": 24,
          "height": 30,
          "marginTop": 20,
          "width": "100%",
        }
      }
      testID="test-input-email-phone"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 26,
      }
    }
  >
    <View
      style={
        Object {
          "flex": null,
          "flexDirection": "column",
          "justifyContent": "flex-end",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#FFF",
              "height": 250,
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              1
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              2
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              3
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              4
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              5
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              6
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              7
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              8
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              9
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Image
              style={
                Array [
                  Object {
                    "height": 28,
                    "width": 28,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#222222",
                    "fontFamily": "HelveticaNeue",
                    "fontSize": 25,
                    "fontWeight": "400",
                    "textAlign": "center",
                  },
                  undefined,
                ]
              }
            >
              0
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onLayout={[Function]}
            onPressIn={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onRippleAnimation={[Function]}
            onStartShouldSetResponder={[Function]}
            pointerEvents="box-only"
            rippleCentered={false}
            rippleColor="#000"
            rippleDuration={450}
            rippleFades={true}
            rippleOpacity={0.3}
            rippleSequential={false}
            rippleSize={250}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomColor": "#e8e8e8",
                "borderBottomWidth": 0,
                "borderRightColor": "#e8e8e8",
                "borderRightWidth": 0,
                "flex": 1,
                "flexDirection": "column",
                "justifyContent": "center",
              }
            }
          >
            <Image
              source={1}
              style={
                Array [
                  Object {
                    "height": 28,
                    "width": 28,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "left": 0,
                    "overflow": "hidden",
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "borderRadius": 0,
                  },
                  Object {},
                ]
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 20,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#00d650",
            "borderRadius": 30,
            "fontFamily": "GothamMed",
            "height": 81.42857142857143,
            "justifyContent": "center",
            "opacity": 1,
            "width": "90%",
          }
        }
        testID="test-button-phone-next"
      >
        <Text
          style={
            Array [
              Object {
                "color": "white",
                "fontFamily": "GothamMed",
                "fontSize": 30,
              },
              Object {
                "color": "rgba(244, 244, 244, 0.5)",
              },
            ]
          }
        >
          Next
        </Text>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
